### DISCUSSION TITLE
Use Livewire to package development

### DISCUSSION THREAD

**Discussion by reziamini** [livewire/livewire#1921] [4. General]
Hey, I'm working on a package and I wanna develop it by Livewire.
now I have an error.
My package structure : 

```
├── Admin Panel
│   ├── Http
│   │   ├── Livewie 
│   │   │   ├── Todo
│   │   │   │   ├── Create
```

In my provider I have registered Create component with `Livewire::component` but when I see my page
It gives an error 
`Unable to find component: [admin-panel.http.livewire.todo.create]`
How can I reference my view to Livewire component in My package ?
now Livewire tries to find a component with directory structure but I register components with Livewire Facade in my SP 

Also my view structure after publish :
```
├── resources
│   ├── view
│   │   ├── vendor 
│   │   │   ├── admin
│   │   │   │   ├── livewire
│   │   │   │   │   ├── todo
│   │   │   │   │   │   ├── create
```

My render method in Create.php Component : 
```php
public function render()
    {
        return view('admin::livewire.todo.create')
            ->layout('admin::layouts.app');
    }
```

**Comment by joshhanley** [comment]
You will need to register the views in your service provider alongside the component registration, see here for registering blade views https://laravel.com/docs/8.x/packages#views

Have a look at this package's service provider, it might help getting them configured https://github.com/tanthammar/tall-forms/blob/master/src/FormServiceProvider.php

Hope this helps!

**Reply by reziamini** [reply]
I've registered my views.
If I set component name to `admin-panel.http.livewire.todo.lists` in my SP it works and I think It's in depending on Namespace.
But I want to set a short name like `admin.todo.list` how can I do this ?

**Reply by joshhanley** [reply]
Hmm well in your package, don't nest it that deep in the folder structure. In your package you should be able to just have `Admin Panel > Todo > Create`, so when you map it, it comes out as `admin-panel::todo.create`

Or map the path differently.

Can you post what you have in your service provider currently? That will make it easier to see what happening and how you might change it.
